### ISSUE TITLE
Convert ImageUploader to use AssetManagerAndQuarantinedFileStorage

### ISSUE THREAD

**Issue by chrisroos** [alphagov/asset-manager#325]
So that we start uploading new `ImageUploader` images to Asset Manager as well as storing them on disk.

**Comment by chrislo** [comment]
I've [created a branch](https://github.com/alphagov/whitehall/compare/use-asset-manager-and-nfs-for-image-uploader) which uses the `asset_manager_and_quarantined_file_storage` as the storage engine for `ImageUploader`. With this change all existing tests pass. I want to convince myself that things are actually working as expected though for the 8 models that use this uploader so am looking at the integration tests.

**Comment by chrislo** [comment]
I've closed alphagov/whitehall#3589 for now as replacing an image with a file of the same name will cause an exception. #336 addresses that, and if it is merged I think the former PR can also be merged without change.

**Comment by chrislo** [comment]
I merged https://github.com/alphagov/whitehall/pull/3602 which should address this issue. We'll want to check that new assets are being added to Asset Manager when this is deployed.

**Comment by chrisroos** [comment]
@chrislo: I'm reopening this issue to remind us to check the behaviour once this has been deployed. Please feel free to close it again if you have something else to remind us to check the behaviour.

**Comment by chrislo** [comment]
This has now been deployed to Integration. I performed the following manual tests in Whitehall and an Asset Manager rails console on `backend-1`:

Creating
========

-   I [created a new Person](https://whitehall-admin.integration.publishing.service.gov.uk/government/admin/people/) with an image
-   assigned them to a role (to ensure full version of published person page is rendered)
-   Verified that a [thumbnail image](https://assets-origin.integration.publishing.service.gov.uk/government/uploads/system/uploads/person/image/3323/s216_big-cheese.960x640.jpg%0A) was available on the "live site"
-   Verified that a [thumbnail image](https://whitehall-admin.integration.publishing.service.gov.uk/government/uploads/system/uploads/person/image/3323/s300_big-cheese.960x640.jpg%0A) was also available on the admin site
-   Verified that 7 new WhitehallAssets were created in Asset Manager `WhitehallAsset.where(legacy_url_path: /3323/).count #=> 7`

Replacing
=========

-   Replaced the person image with a new one
-   Verified the [new thumbnail](https://assets-origin.integration.publishing.service.gov.uk/government/uploads/system/uploads/person/image/3323/s216_minister-of-funk.960x640.jpg%0A) was available on the "live site"
-   Verified the [new thumbnail](https://whitehall-admin.integration.publishing.service.gov.uk/government/uploads/system/uploads/person/image/3323/s300_minister-of-funk.960x640.jpg%0A) was availbe on the admin site
-   14 WhitehallAssets in asset manager `WhitehallAsset.where(legacy_url_path: /3323/).count #=> 14`

Replacing with an asset of the same name
========================================

-   Replaced the person image with a file with different content but the same filename
-   Verified the [new thumbnail](https://whitehall-admin.integration.publishing.service.gov.uk/government/uploads/system/uploads/person/image/3323/s300_minister-of-funk.960x640.jpg%0A) was available on the admin site (cache has to be hard refreshed)
-   Verified the [new thumbail](https://assets-origin.integration.publishing.service.gov.uk/government/uploads/system/uploads/person/image/3323/s216_minister-of-funk.960x640.jpg%0A) was available on the live site
-   Verified that there were 7 assets marked as deleted in Asset Manager `WhitehallAsset.unscoped.where(legacy_url_path: /3323/).count #=> 21`, `WhitehallAsset.where(legacy_url_path: /3323/).count #=> 14`

Removing
========

-   Removed person from assigned role
-   Verified latest image marked as deleted in Asset Manager and previously replaced image was not deleted `WhitehallAsset.unscoped.where(legacy_url_path: /3323/).count #=> 21`, `WhitehallAsset.where(legacy_url_path: /3323/).count #=> 7`.

**Comment by chrislo** [comment]
I'm happy that Integration has the correct behaviour with regards to People images (which are backed by `ImageUploader`). I'm going to ask for `release_13133` to be deployed to staging and production.

**Comment by chrislo** [comment]
`release_13133` of Whitehall has been deployed to staging and production. There was an issue on staging due to the bearer token that Whitehall has to connect to Asset Manager having been changed (causing [GdsApi::HTTPUnauthorized](https://sentry.io/govuk/app-whitehall/issues/401637759/events/10602377768/) exceptions to be raised from the `AssetManagerCreateWhitehallAssetWorker` worker). I'm not entirely sure why these tokens had changed, but I believe there are some details in https://github.com/alphagov/govuk-secrets/pull/141 (I can't see this PR). 

Once `govuk-secrets` was updated there were no more exceptions and we deployed to production.
